cocos/math/CCGeometry.cpp

```diff
@@ -188,6 +188,41 @@ bool Rect::intersectsRect(const Rect& rect) const
              rect.getMaxY() <      getMinY());
 }
 
+bool Rect::intersectsCircle(const cocos2d::Vec2 &center, float radius) const
+{
+    Vec2 rectangleCenter = Vec2((origin.x + size.width / 2),
+                                (origin.y + size.height / 2));
+    
```

Comment:
Use `Vec2 rectangleCenter((origin.x + size.width / 2), (origin.y + size.height / 2))` to optimize it?